### PR TITLE
sdk/log: Package documentation

### PR DESCRIPTION
**Pull request by pellared**
Fixes https://github.com/open-telemetry/opentelemetry-go/issues/5065

Based on https://pkg.go.dev/go.opentelemetry.io/otel/sdk/metric

### CODE REVIEW

**Review by dmathieu — APPROVED** [review]

File: sdk/log/doc.go [hunk 1]
```diff
@@ -1,9 +1,36 @@
...
+
+The entry point for the log package is [NewLoggerProvider].
+[LoggerProvider] is the object that all Bridge API calls use to create
+Loggers, and ultimately emit log records.
+Also, it is an object that should be used to
+control the life-cycle (start, flush, and shutdown) of the Logs SDK.
+
+A LoggerProvider needs to be configured to process the log records, this is
+done by configuring it with a [Processor] implementation using [WithProcessor].
+The log package provides [BatchProcessor] and [SimpleProcessor]
```
Comment (MrAlias) [inline sdk/log/doc.go:20]:
```suggestion
The log package provides the [BatchProcessor] and [SimpleProcessor]
```

File: sdk/log/doc.go [hunk 2]
```diff
@@ -1,9 +1,36 @@
...
+that are configured with an [Exporter] implementation and
```
Comment (MrAlias) [inline sdk/log/doc.go:21]:
```suggestion
that are configured with an [Exporter] implementation which
```

File: sdk/log/doc.go [hunk 3]
```diff
@@ -1,9 +1,36 @@
...
+exports the log records to given destination. See
+[go.opentelemetry.io/otel/exporters] for exporters that can be used as
```
Comment (MrAlias) [inline sdk/log/doc.go:23]: What exporters can be used as a processor?

File: sdk/log/doc.go [hunk 4]
```diff
@@ -1,9 +1,36 @@
...
+or with these Processors.
+
+The data generated by a LoggerProvider needs to include information about its
+origin. A LoggerProvider needs to be configured with a Resource, using the
```
Comment (MrAlias) [inline sdk/log/doc.go:27]:
```suggestion
origin. A LoggerProvider needs to be configured with a Resource, using
```

or

```suggestion
origin. A LoggerProvider needs to be configured with a Resource, by using
```

**Review by MrAlias — APPROVED** [review]

File: sdk/log/doc.go (hunk 3 above)
Comment (pellared) [reply sdk/log/doc.go]: https://github.com/open-telemetry/opentelemetry-go/pull/5609/commits/f61e71ab43d90ccffb993ee56bdb3cc90ff76943

File: sdk/log/doc.go [hunk 5]
```diff
@@ -1,9 +1,36 @@
...
+Loggers, and ultimately emit log records.
+Also, it is an object that should be used to
+control the life-cycle (start, flush, and shutdown) of the Logs SDK.
+
+A LoggerProvider needs to be configured to process the log records, this is
+done by configuring it with a [Processor] implementation using [WithProcessor].
+The log package provides the [BatchProcessor] and [SimpleProcessor]
+that are configured with an [Exporter] implementation which
+exports the log records to given destination. See
+[go.opentelemetry.io/otel/exporters] for exporters that can be use with these
```
Comment (MrAlias) [inline sdk/log/doc.go:23]:
```suggestion
[go.opentelemetry.io/otel/exporters] for exporters that can be used with these
```

**Review by dashpole — APPROVED** [review]

**Pull request merged**